test/integration/host_test.rb

```diff
@@ -117,6 +117,8 @@ def assert_interface_change(change, &block)
           assert modal.find('.interface_provision').checked?, "Provision checkbox is checked"
 
           modal.find(:button, "Cancel").click
+          # wait until modal is closed
+          page.has_css?('#interfaceModal', :visible => false)
```

Comment:
AFAIK capybara should wait until such element exist or time out. The table already exist by the time the modal is being closed so it doesn't block and directly checks the content (my assumption).